service/src/main/java/bio/terra/user/app/controller/ProfileApiController.java

```diff
@@ -19,32 +21,44 @@ public class ProfileApiController implements ProfileApi {
   private final HttpServletRequest servletRequest;
   private final ProfileService profileService;
   private final ProfileSamUserFactory userFactory;
+  private final SamService samService;
 
   @Autowired
   public ProfileApiController(
       HttpServletRequest servletRequest,
       ProfileService profileService,
-      ProfileSamUserFactory userFactory) {
+      ProfileSamUserFactory userFactory,
+      SamService samService) {
     this.servletRequest = servletRequest;
     this.profileService = profileService;
     this.userFactory = userFactory;
+    this.samService = samService;
   }
 
   @Override
-  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path) {
-    profileService.setProperty(getUser(), parsePath(path), body.getValue());
+  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path, String userEmail) {
```

Comment:
We have a convention that the first thing controller does, is check access control. [Example.](https://cs.github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java#L298)

[Here's an example of checking if requester is SAM admin.](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/app/controller/AdminApiController.java#L36-L39)

So you would have:

```
if (StringUtils.isEmpty(userEmail)) {
      AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
    SamRethrow.onInterrupted(
        () -> getSamService().checkAdminAuthz(userRequest),
        "check whether the user has admin access");
}
```

Reply:
Fair enough, though that's pretty much the purpose of `getProfileUser`, just merged into one sam call rather than two by checking auth and getting the user id at once